### PR TITLE
Create template

### PR DESCRIPTION
**Pull request by Insprill**
Mostly the same as the old BepInEx template, with the main difference being the `package.ps1` script has been updated not to require modification, and Harmony patching is now included in the Load method.

### CODE REVIEW

File: MOD_NAME/MOD_NAME.csproj [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+	<!-- Unity -->
+	<ItemGroup>
+		<Reference Include="UnityEngine"/>
+		<Reference Include="UnityEngine.CoreModule"/>
+	</ItemGroup>
+
+	<!-- Mod Loader -->
+	<ItemGroup>
+		<PackageReference Include="UnityModManager" Version="0.27.2"/>
```
Comment (fauxnik) [inline MOD_NAME/MOD_NAME.csproj:23]: This is cool! I guess this lets us avoid having to include it in the `Directory.Build.targets` file?

Is 0.27.3 available? Not sure if it changes much in this context, but my understanding is that that's the version we need for everything to work the way we asked it to be changed to.

File: info.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "Id": "MOD_NAME",
+  "Version": "0.1.0",
+  "DisplayName": "MOD_NAME",
+  "Author": "Your Name",
```
Comment (fauxnik) [inline info.json]: I can't figure out how to suggest edits like you did for me, but I'd suggest using `MOD_AUTHOR` here instead.

File: info.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "Id": "MOD_NAME",
+  "Version": "0.1.0",
```
Comment (fauxnik) [inline info.json]: I think this should be `1.0.0`.

File: info.json
Comment (fauxnik) [inline info.json:1]: I've typically seen this filename capitalized in the context of UMM mods, i.e. `Info.json`. Do you know whether or not it matters?

**Review by fauxnik — COMMENTED** [review]
Thanks for picking this up!

File: MOD_NAME/MOD_NAME.csproj (hunk 1 above)
Comment (Insprill) [reply MOD_NAME/MOD_NAME.csproj:23]: > This is cool! I guess this lets us avoid having to include it in the Directory.Build.targets file?

Yup!

> Is 0.27.3 available?

0.27.2 is the latest on NuGet. The only difference between the two is in Derail Valley's configuration.

File: info.json
Comment (Insprill) [reply info.json:1]: It matters on case-sensitive file systems, like EXT-4 and BTRFS.

File: info.json (hunk 2 above)
Comment (Insprill) [reply info.json]: You can suggest edits by doing

```
```suggestion
New code here
```

File: info.json
Comment (fauxnik) [reply info.json:1]: I know there are a few folks who play on Linux. Would it matter for them? I'm not familiar with what they use to run a Windows game on Linux, but I imagine their filesystems are probably EXT-4 or some such.

Comment (Insprill) [reply info.json:1]: It does matter, yes. Almost every Linux system uses EXT-3/4, BTRS, or ZFS, all of which are case-sensitive.

Comment (fauxnik) [reply info.json:1]: I see. So then, which casing is correct?

Comment (Insprill) [reply info.json:1]: Lowercase is correct.

**Review by fauxnik — APPROVED** [review]

**Pull request merged**